src/main/java/ca/corefacility/bioinformatics/irida/repositories/filesystem/IridaFileStorageAzureServiceImpl.java

```diff
@@ -0,0 +1,176 @@
+package ca.corefacility.bioinformatics.irida.repositories.filesystem;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.util.Date;
+import java.util.zip.GZIPInputStream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.BlobServiceClientBuilder;
+import com.azure.storage.blob.models.BlobStorageException;
+
+/**
+ * Component implementation of file utitlities for azure storage
+ */
+@Component
+public class IridaFileStorageAzureServiceImpl implements IridaFileStorageService {
```

Comment:
Really like what's happening with this class.  That's exactly what I'd hope to see with the interface and multiple implementations.

Reply:
Thanks :)